packages/apps/spaces/app/organization/[id]/src/components/Timeline/preview/context/TimelineEventPreviewContext.tsx

```diff
@@ -62,25 +69,60 @@ export const TimelineEventPreviewContextContextProvider = ({
     setModalContent(content);
   };
 
-  const handleCloseModal = () => {
-    if (!isModalOpen) return;
+  const handleDeleteParams = () => {
     const params = new URLSearchParams(searchParams?.toString() ?? '');
     params.delete('events');
-    setIsModalOpen(false);
-    setModalContent(null);
     setLastActivePosition({
       ...lastActivePosition,
       [id]: params.toString(),
     });
     router.push(`?${params}`);
   };
 
+  const handleCloseModal = () => {
+    if (!isModalOpen) return;
+    setIsModalOpen(false);
+    setModalContent(null);
+    handleDeleteParams();
+  };
+
+  const getModalContentFromServer = async (id: string) => {
+    try {
+      const result = await client.request<GetTimelineEventQuery>(
```

Comment:
Is there a reason why we could'nt use a react-query hook?

Reply:
yes, it's called conditionally

Reply:
I see. The react-query docs offers this way of doing dependent/conditional queries [link](https://tanstack.com/query/v3/docs/react/guides/dependent-queries)

Reply:
I know it seems counter intuitive but we loose all the caching, loading, error, etc etc functionalities otherwise.

Reply:
Thanks, I was not aware of this possibility. The naming would be more intuitive if they'd use "conditional" instead of  "dependent". 

I'll refactor this code accordingly

Reply:
One more thing which for me would make usage of the hooks better. Can we not call the hook when the modal mounts instead of calling it conditionally in the context?

Reply:
it would still need to be called conditionally

Reply:
this code is not checking if modal got open and then loads data. Ifs checking if the modalData was passed and if it was not passed then getting it from the be